src/store/sagas/bhr.ts

```diff
@@ -379,6 +380,23 @@ function* getAppImageWorker({ payload }) {
     }
 
     const recoveryKeySigner = setupRecoveryKeySigningKey(primaryMnemonic);
+
+    if (!appImage.wallets.length && !allVaultImages.length) {
+      // recreate first wallet
+      const defaultWallet: NewWalletInfo = {
+        walletType: WalletType.DEFAULT,
+        walletDetails: {
+          name: 'Mobile Wallet',
+          description: '',
+          transferPolicy: {
+            id: uuidv4(),
+            threshold: 0,
+          },
+          instanceNum: 0,
+        },
```

Comment:
Ideally it should explicitly specify the derivation used as BIP84 by adding this:
derivationConfig: {
            path: WalletUtilities.getDerivationPath(
              false,
              config.NETWORK_TYPE,
              0,
              DerivationPurpose.BIP84
            ),
            purpose: DerivationPurpose.BIP84,
          },